package/MDAnalysis/analysis/density.py

```diff
@@ -383,7 +384,18 @@ def __init__(self, atomgroup, delta=1.0,
         self._ydim = ydim
         self._zdim = zdim
 
+    def _checkinput(self):
+        if len(self._atomgroup) == 0:
+            msg = ("No atoms in selection over whole trajectory")
+            warnings.warn(msg)
+            logger.warning(msg)
+            return 1
+        else:
+            return 0
+
     def _prepare(self):
+        if _checkinput(self) == 1:
+            return
```

Comment:
Couple of points to consider here;

1. Is this the right place to be putting this? Given the context of the original issue, and the need to make it so that the user defined grid still works even if there are no atoms in the frame, would placing this elsewhere be more appropriate?

2. Doing a `return` post-warning is a non-safe way to handle things. It's worth considering what the difference between a "warning" and an "error" is. If you're issuing a warning, you are telling users that non-expected behaviour is likely to occur (and why) but that the code will carry on. If you raise an error, you are telling users that something wrong/unexpected has occurred and you are aborting further code execution. In this case, if you are attempting to abort execution (as you are via `return`), then an error (of appropriate type) should be raised instead.

3. I think the way this is written probably needs to be changed, but it's worth mentioning anyways. Doing a return equality check on method return that essentially returns a `Boolean` is a non-pythonic way to handle things. If you're checking for `True`/`False`, the best way to do things is to have the method you're calling return `True`/`False` and then just do:
```python
if method(input):
    true_condition
else:
    false_condition
```